### PR TITLE
Modifying forward function to better illustrate the picture

### LINKED ISSUE #2788: NN tutorial - picture and code are confusing at first glance
### 🚀 Descirbe the improvement or the new tutorial

Link to tutorial: https://pytorch.org/tutorials/beginner/blitz/neural_networks_tutorial.html  

Trying to learn and understand PyTorch I was utterly confused when comparing the picture and the corresponding code

    def __init__(self):
        super(Net, self).__init__()
        # 1 input image channel, 6 output channels, 5x5 square convolution
        # kernel
        self.conv1 = nn.Conv2d(1, 6, 5)
        self.conv2 = nn.Conv2d(6, 16, 5)
        # an affine operation: y = Wx + b
        self.fc1 = nn.Linear(16 * 5 * 5, 120)  # 5*5 from image dimension
        self.fc2 = nn.Linear(120, 84)
        self.fc3 = nn.Linear(84, 10)

where are the subsampling layers? I first thought you simply dropped them and used an outdated picture. Only much later I saw that they are included in the forward function. Maybe something like this would be better:

     def __init__(self):
        # input channel 32x32 pixel
        super(Net, self).__init__()
        # convolution layer C1: 1 input image channel, 6 output channels, 5x5 square convolution => 28x28 pixel
        self.conv1 = nn.Conv2d(1, 6, 5)
        # subsampling layer S2: 2x2 grid, purely functional, does not keep any parameter => 14x14 pixel
        # convolution layer C3: 6 input channels, 16 output channels,  5x5 square convolution => 10x10 pixel
        self.conv2 = nn.Conv2d(6, 16, 5)
        # subsampling layer S4: 2x2 grid, purely functional, does not keep any parameter => 5x5 pixel
        # flatten operation: purely functional, 2D picture => 1D array
        # fully connected layer F4: 160 inputs, 120 outputs, uses RELU activation function
        self.fc1 = nn.Linear(16 * 5 * 5, 120)  # 5*5 from image dimension
        # fully connected layer F5: 120 inputs, 84 outputs, uses RELU activation function
        self.fc2 = nn.Linear(120, 84)
        # gaussian layer F6: 84 inputs, 10 outputs
        self.fc3 = nn.Linear(84, 10)

### Existing tutorials on this topic

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jmarintur**
Fixes #2788

## Description
Improving forward function to better illustrate the picture in the neural network tutorial.

## Checklist
- [x] The issue that is being fixed is referred in the description 
- [x] Only one issue is addressed in this pull request
- [ ] Labels from the issue that this PR is fixing are added to this pull request
- [x] No unnecessary issues are included into this pull request.

### CODE REVIEW

File: beginner_source/blitz/neural_networks_tutorial.py [hunk 1]
```diff
@@ -55,16 +55,30 @@ def __init__(self):
...
-        # If the size is a square, you can specify with a single number
-        x = F.max_pool2d(F.relu(self.conv2(x)), 2)
-        x = torch.flatten(x, 1) # flatten all dimensions except the batch dimension
-        x = F.relu(self.fc1(x))
-        x = F.relu(self.fc2(x))
-        x = self.fc3(x)
-        return x
+    def forward(self, input):
+        # Convolution layer C1: 1 input image channel, 6 output channels,
+        # 5x5 square convolution, it uses RELU activation function => 6x28x28 tensor
```
Comment (albanD) [inline beginner_source/blitz/neural_networks_tutorial.py:60]:
```suggestion
        # 5x5 square convolution, it uses RELU activation function and outputs a Tensor with size (6, 28, 28)
```

This feels clearer to me? If you agree we could update this and the ones below.

File: beginner_source/blitz/neural_networks_tutorial.py [hunk 2]
```diff
@@ -55,16 +55,30 @@ def __init__(self):
...
+        c1 = F.relu(self.conv1(input))
+        # Subsampling layer S2: 2x2 grid, purely functional,
+        # it does not keep any parameter => 6x14x14 tensor
```
Comment (albanD) [inline beginner_source/blitz/neural_networks_tutorial.py:63]:
```suggestion
        # this layer does not have any parameter => 6x14x14 tensor
```

Comment (albanD) [reply beginner_source/blitz/neural_networks_tutorial.py:63]: Same comment for other occurrences below

File: beginner_source/blitz/neural_networks_tutorial.py (hunk 1 above)
Comment (albanD) [reply beginner_source/blitz/neural_networks_tutorial.py:60]: Maybe even `(B, 6, 28, 28)` if you want to show how the batch appears and how it is impacted by the flatten op below.

File: beginner_source/blitz/neural_networks_tutorial.py [hunk 3]
```diff
@@ -55,16 +55,30 @@ def __init__(self):
...
+        s2 = F.max_pool2d(c1, (2, 2))
+        # Convolution layer C3: 6 input channels, 16 output channels,
+        # 5x5 square convolution, it uses RELU activation function => 16x10x10 tensor
+        c3 = F.relu(self.conv2(s2))
+        # Subsampling layer S4: 2x2 grid, purely functional,
+        # it does not keep any parameter => 16x5x5 tensor
+        s4 = F.max_pool2d(c3, 2)
+        # Flatten operation: purely functional, 2D tensor => 1D tensor
```
Comment (albanD) [inline beginner_source/blitz/neural_networks_tutorial.py:71]: Input is actually 3D here: C H W
With the caveat about batch dim as mentioned above

**Review by albanD — COMMENTED** [review]
Sounds good! Small edit suggestions inline

**Comment by jmarintur** [comment]
Thank very much for your suggestions, please let me know what you think about the changes I did following them. By the way, I tried to add you as co-author, but I didn't find out what was your email @albanD.

**Review by albanD — APPROVED** [review]
np np no need for cc author !

cc @svekars this looks good content-wise!
